lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js

```diff
@@ -208,19 +210,43 @@ function handleLogs() {
       }
     }
 
-    const destinationArn = {
-      op: 'replace',
-      path: '/accessLogSettings/destinationArn',
-      value: `arn:aws:logs:${region}:${this.accountId}:log-group:${logGroupName}`,
-    };
-    const format = {
+    let enableAccessLogs = logs.enableAccessLogging;
+    if (enableAccessLogs === undefined) {
+      enableAccessLogs = true;
+    }
+
+    if (enableAccessLogs) {
+      const destinationArn = {
+        op: 'replace',
+        path: '/accessLogSettings/destinationArn',
+        value: `arn:aws:logs:${region}:${this.accountId}:log-group:${logGroupName}`,
+      };
+      const format = {
+        op: 'replace',
+        path: '/accessLogSettings/format',
+        value: logFormat,
+      };
+
+      operations.push(destinationArn, format);
+    } else {
+      // this is required to remove any existing log setting
+      operations.push({
+        op: 'remove',
+        path: '/accessLogSettings',
+      });
+    }
+
+    let enableDataTracing = logs.dataTrace;
+    if (enableDataTracing === undefined) {
+      enableDataTracing = true;
+    }
+    operations.push({
       op: 'replace',
-      path: '/accessLogSettings/format',
-      value: logFormat,
-    };
-    dataTrace = { op: 'replace', path: '/*/*/logging/dataTrace', value: 'true' };
-    logLevel = { op: 'replace', path: '/*/*/logging/loglevel', value: level };
-    operations = [destinationArn, format, dataTrace, logLevel];
+      path: '/*/*/logging/dataTrace',
+      value: (!!enableDataTracing).toString(),
```

Comment:
nit: more self explanatory (less hacky) might be: `value: String(Boolean(enableDataTracing))`

Reply:
Sure, I'll change it. Wasn't too sure what the idiomatic way to do this in js was, thanks for the suggestion.